infra/aws/terraform/audit-account/guarduty.tf

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+resource "aws_guardduty_detector" "default" {
+  provider = aws.audit
+  enable   = true
+
+  tags = merge({
+    Env = "Audit"
+  }, var.tags)
+}
```

Comment:
What regions will we have this in?

Reply:
we want all the regions to be covered.

Reply:
I think we'll need to repeat this for each region where we permit AWS infrastructure.

Reply:
> Amazon GuardDuty is a region-based service. To activate GuardDuty on another AWS regions, repeat steps &hellip; for each AWS regions.

&mdash;AWS

Reply:
This typically looks like a provider per region. Maybe also SCPs to block use of regions we're not covering.

Plus simple (detector only, _no_ `aws_guardduty_organization_configuration`) GuardDuty configs in the management account only, for any regions that are:
- enabled by default (see https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_examples_aws-enable-disable-regions.html for context)
- _excluded_ by an SCP

Those simple detectors are because the region-exclude SCP won't cover the management account. We can use the SCP exclusion to infer that there's no infrastructure in leaf accounts for the excluded regions, and hence no possibility of problematic cross-account access.

Reply:
I think it's better we create a tf module for guardduty and enable it for all the regions. If the it's a region-based service, how the aggregation is happening from the org member accounts to the delegator administrator account for guardduty ?

Reply:
Each region has its own aggregation target, and those targets are all in the delegated administrator account.

You (ie, we) can build your own cross-region aggregation into another service - which isn't trivial.

Reply:
Implementation will done in a separate PR.